src/scenes/Activities/scenes/EventHistory/index.js

```diff
@@ -0,0 +1,249 @@
+            <TableCell
+              key={column.id}
+              numeric={column.numeric}
+              sortDirection={orderBy === column.id ? order : false}
+            >
+              <Tooltip
+                title={<FormattedMessage id="str.sort" default="Sort" />}
+                enterDelay={Config.intervals.tooltipDelay}
+                placement={column.numeric ? 'bottom-end' : 'bottom-start'}
+              >
+                <TableSortLabel
+                  active={orderBy === column.id}
+                  direction={order}
+                  onClick={this.createSortHandler(column.id)}
+                >
+                  <FormattedMessage id={column.name} default={column.nameDefault} />
+                </TableSortLabel>
+              </Tooltip>
+            </TableCell>
+          ))}
+        </TableRow>
+      </TableHead>
+    );
+  }
+
+  getTableRows(transactions) {
+    const { classes } = this.props;
+
+    return (
+      <TableBody>
+        {_.map(transactions, (transaction, index) => (
+          transaction.topic ?
+            <TableRow key={transaction.txid} selected={index % 2 === 1}>
+              <TableCell>
+                {getShortLocalDateTimeString(transaction.createdTime)}
+              </TableCell>
+              <TableCell>
+                {transaction.type}
+              </TableCell>
+              <TableCell>
+                {transaction.topic ? transaction.topic.name : null}
+              </TableCell>
+              <TableCell>
+                {transaction.token}
+              </TableCell>
+              <TableCell numeric>
+                {transaction.amount}
+              </TableCell>
+              <TableCell numeric>
+                {transaction.fee}
+              </TableCell>
+              <TableCell>
+                {transaction.status}
+              </TableCell>
+              <TableCell>
+                <Link to={this.getEventURL(transaction.topic.address, transaction.oracleAddress)} className={classes.viewEventLink}>
+                  <FormattedMessage id="eventHistory.viewEvent" default="View Event" />
+                </Link>
+              </TableCell>
+            </TableRow> : null
+        ))}
+      </TableBody>
+    );
+  }
+
+  createSortHandler = (property) => (event) => {
+    this.handleSorting(event, property);
+  };
+
+  handleSorting(event, property) {
+    const orderBy = property;
+    let order = 'desc';
+
+    if (this.state.orderBy === property && this.state.order === 'desc') {
+      order = 'asc';
+    }
+
+    const transactions = order === 'desc'
+      ? this.state.transactions.sort((a, b) => (b[orderBy] < a[orderBy] ? -1 : 1))
+      : this.state.transactions.sort((a, b) => (a[orderBy] < b[orderBy] ? -1 : 1));
+
+    this.setState({ transactions, order, orderBy });
+  }
+
+  getEventURL(topicAddress, oracleAddress) {
+    // TODO: NEED TO REROUTE TO TOPIC PAGE IS ITS IN WITHDRAW STATUS
+    return `/oracle/${topicAddress}/${oracleAddress}`;
```

Comment:
also need to handle `Withdraw` txs too. Withdraw txs are actually done on the TopicEvent itself (not an Oracle).